lib/patches/aws-sdk-core/seahorse_client_net_http_pool_patch.rb

```diff
@@ -0,0 +1,44 @@
+# Autoload the connection pool
+Seahorse::Client::NetHttp::ConnectionPool
+
+module Seahorse
+  module Client
+    module NetHttp
+      class ConnectionPool
+        def start_session endpoint
+
+          endpoint = URI.parse(endpoint)
+
+          args = []
+          args << endpoint.host
+          args << endpoint.port
+          args << http_proxy.host
+          args << http_proxy.port
+          args << (http_proxy.user && CGI::unescape(http_proxy.user))
+          args << (http_proxy.password && CGI::unescape(http_proxy.password))
```

Comment:
I think this makes sense, monkey patch the single method instead of bringing along 2 method changes from the PR.